cmake/crc32c.cmake

```diff
@@ -78,46 +78,33 @@ check_cxx_source_compiles_with_flags("${ARM64_CRC_CXXFLAGS}" "
   " HAVE_ARM64_CRC32C
 )
 
-add_library(crc32c_common INTERFACE)
-target_compile_definitions(crc32c_common INTERFACE
+add_library(crc32c STATIC EXCLUDE_FROM_ALL
+  ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c.cc
+  ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_portable.cc
+)
+target_compile_definitions(crc32c PRIVATE
   HAVE_BUILTIN_PREFETCH=$<BOOL:${HAVE_BUILTIN_PREFETCH}>
   HAVE_MM_PREFETCH=$<BOOL:${HAVE_MM_PREFETCH}>
   HAVE_STRONG_GETAUXVAL=$<BOOL:${HAVE_STRONG_GETAUXVAL}>
   BYTE_ORDER_BIG_ENDIAN=$<STREQUAL:${CMAKE_CXX_BYTE_ORDER},BIG_ENDIAN>
   HAVE_SSE42=$<BOOL:${HAVE_SSE42}>
   HAVE_ARM64_CRC32C=$<BOOL:${HAVE_ARM64_CRC32C}>
 )
-target_link_libraries(crc32c_common INTERFACE
-  core_interface
-)
-
-add_library(crc32c STATIC EXCLUDE_FROM_ALL
-  ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c.cc
-  ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_portable.cc
-)
 target_include_directories(crc32c
   PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/crc32c/include>
 )
-target_link_libraries(crc32c PRIVATE crc32c_common)
+target_link_libraries(crc32c PRIVATE core_interface)
 set_target_properties(crc32c PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
 
 if(HAVE_SSE42)
-  add_library(crc32c_sse42 STATIC EXCLUDE_FROM_ALL
-    ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_sse42.cc
-  )
-  target_compile_options(crc32c_sse42 PRIVATE ${SSE42_CXXFLAGS})
-  target_link_libraries(crc32c_sse42 PRIVATE crc32c_common)
-  set_target_properties(crc32c_sse42 PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
-  target_link_libraries(crc32c PRIVATE crc32c_sse42)
+  set(srcs ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_sse42.cc)
+  target_sources(crc32c PRIVATE ${srcs})
+  set_property(SOURCE ${srcs} PROPERTY COMPILE_OPTIONS ${SSE42_CXXFLAGS})
 endif()
 
 if(HAVE_ARM64_CRC32C)
-  add_library(crc32c_arm64 STATIC EXCLUDE_FROM_ALL
-    ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_arm64.cc
-  )
-  target_compile_options(crc32c_arm64 PRIVATE ${ARM64_CRC_CXXFLAGS})
-  target_link_libraries(crc32c_arm64 PRIVATE crc32c_common)
-  set_target_properties(crc32c_arm64 PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
-  target_link_libraries(crc32c PRIVATE crc32c_arm64)
+  set(srcs ${PROJECT_SOURCE_DIR}/src/crc32c/src/crc32c_arm64.cc)
```

Comment:
Unset the `srcs` variable at the end to avoid polluting the global namespace?

Reply:
Done. Also renamed the variable to `_crc32_src`.